### PR TITLE
Added memoization function in fibonacci

### PR DESCRIPTION
**Pull request by JDeepD**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/fibonacci.py [hunk 1]
```diff
@@ -127,4 +157,5 @@ def fib_binet(n: int) -> list[int]:
     num = 20
     time_func(fib_iterative, num)
     time_func(fib_recursive, num)
+    time_func(fib_memoization(), num)
```
Comment (pandrey2003) [inline maths/fibonacci.py:160]: Without parenthesis :)

**Review by pandrey2003 — CHANGES_REQUESTED** [review]
Looks very well overall, just fix one comment

File: maths/fibonacci.py (hunk 1 above)
Comment (pandrey2003) [reply maths/fibonacci.py:160]: `time_func(fib_memoization, num)`

Comment (pandrey2003) [reply maths/fibonacci.py:160]: Oops, my bad. Forgot one small thing :)

**Review by pandrey2003 — APPROVED** [review]
Everything is fine

**Comment by pandrey2003** [comment]
@JDeepD Looks well for me, ask for review people with write access to repo such as @Kush1101 and @cclauss

**Comment by cclauss** [comment]
Please use the same tests as the other implementations.  Please put the test results in the docstring at the top of the file so that visitors can see at a glance how these algorithms perform.

**Comment by JDeepD** [comment]
> Please use the same tests as the other implementations. 

Like, should I output a list of n-indexed fibonnaci numbers??

**Comment by cclauss** [comment]
Like run the file and paste in the text that the last four lines generate.

**Review by cclauss — APPROVED** [review]
Nice performance numbers!  Thanks for the teamwork on this one.

**Pull request merged**